scripts/factory-common-install/bin/device-vnc-enter-as-frontend

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+set -euf -o pipefail
+common_factory_install_libexec_dir="$("$(dirname "$0")/pkg-nixos-factory-common-install-get-libexec-dir")"
+. "$common_factory_install_libexec_dir/app_current_device_ssh.sh"
+. "$common_factory_install_libexec_dir/app_current_device_store.sh"
+
+vnc_user=""
+vnc_hostname=""
+
+is_vnc_user_empty() {
+  if [ -z $vnc_user ]; then
+    echo "Error: The vnc user is not set"
+    return 2
+  fi
+}
+
+is_vnc_hostname_empty() {
```

Comment:
No longer required once moving the `read_or_prompt` above.

Reply:
Shouldn't we check the variables in case the read_or_prompt_for_current_device fails?

Reply:
As discussed, `read_or_prompt_for_current_device` will fail itself when unable to perform its task and make the whole script fail (because of `set -e`). What I sometimes do to make this work too even when `set -e` is not effective, I do something like: `read_or_prompt_for_current_device || return 1` (which is the equivalent of `set -e`).